blocks/library/embed/index.js

```diff
@@ -73,52 +74,127 @@ registerBlock( 'core/embed', {
 		}
 	},
 
-	edit( { attributes, setAttributes, focus, setFocus } ) {
-		const { url, title, caption } = attributes;
+	edit: class extends wp.element.Component {
+		constructor() {
+			super( ...arguments );
+			this.doServerSideRender = this.doServerSideRender.bind( this );
+			this.state = {
+				html: '',
+				type: '',
+				error: false,
+				fetching: false,
+			};
+			this.noPreview = [
+				'facebook.com',
+			];
+			if ( this.props.attributes.url ) {
+				// if the url is already there, we're loading a saved block, so we need to render
+				this.doServerSideRender();
+			}
+		}
 
-		if ( ! url ) {
-			return (
-				<Placeholder icon="cloud" label={ wp.i18n.__( 'Embed URL' ) } className="blocks-embed">
-					<input type="url" className="placeholder__input" placeholder={ wp.i18n.__( 'Enter URL to embed here...' ) } />
-					<Button isLarge>
-						{ wp.i18n.__( 'Embed' ) }
-					</Button>
-				</Placeholder>
+		doServerSideRender( event ) {
+			if ( event ) {
+				event.preventDefault();
+			}
+			const { url } = this.props.attributes;
+			const api_url = wpApiSettings.root + 'oembed/1.0/proxy?url=' + encodeURIComponent( url ) + '&_wpnonce=' + wpApiSettings.nonce; // eslint-disable-line no-undef
+
+			this.setState( { error: false, fetching: true } );
+			fetch( api_url, {
+				credentials: 'include',
+			} ).then(
+				( response ) => {
+					response.json().then( ( obj ) => {
+						const { html, type } = obj;
+						if ( html ) {
+							this.setState( { html, type } );
```

Comment:
When we do async calls in React components, we need to ensure that the callbacks are not executed if the component has been unmounted before the end of the async process. Which generally means one of these two things:

 - If the promise (or the async process) is abortable, we need to abort it in `componentWillUnmount`
 - If it's not abortable, we may use a boolean in the component instance `this.mounted` and do something like `if ( this.mounted) return;` at the beginning of the async callback.

(Aborting is better when possible)